core/src/main/scala/org/apache/spark/util/logging/FileAppender.scala

```diff
@@ -76,7 +76,11 @@ private[spark] class FileAppender(inputStream: InputStream, file: File, bufferSi
           }
         }
       } {
-        closeFile()
+        try {
+          inputStream.close()
```

Comment:
It shouldn't be closed here, because the lifecycle of this stream is outside of this class. In practice, the streams used with this class are closed on error anyway.